json.go

```diff
@@ -0,0 +1,67 @@
+package ff
+
+import (
+	"encoding/json"
+	"io"
+	"strconv"
+
+	"github.com/pkg/errors"
+)
+
+// JSONParser is a parser for config files in JSON format. Input should be
+// an object. The object's keys are treated as flag names, and the object's
+// values as flag values. If the value is an array, the flag will be set
+// multiple times.
+func JSONParser(r io.Reader, set func(name, value string) error) error {
+	var m map[string]interface{}
+	d := json.NewDecoder(r)
+	// Must set UseNumber for stringifyValue to work
+	d.UseNumber()
+	err := d.Decode(&m)
+	if err != nil {
+		return errors.Wrap(err, "error parsing JSON config")
+	}
```

Comment:
Nit: just to make this a bit tighter...

```go
d := json.NewDecoder(r)
d.UseNumber() // stringify expects json.Numbers
var m map[string]interface{}
if err := d.Decode(&m); err != nil {
    return ParseError{Err: err}
}
```

Reply:
If I'm creating a public facing error type, then yeah, this should probably be in a new package.